pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -140,17 +143,101 @@ func readConfig(config io.Reader) (Config, error) {
 	return cfg, err
 }
 
+// parseMetadataUUID reads JSON from OpenStack metadata server and parses
+// instance ID out of it.
+func parseMetadataUUID(jsonData []byte) (string, error) {
+	// We should receive an object with { 'uuid': '<uuid> } and couple of other
+	// properties (which we ignore).
+
+	var obj map[string]interface{}
```

Comment:
Fyi, this can be done more easily (and idiomatically) by Unmarshalling into a struct - see https://github.com/kubernetes/kubernetes/pull/17041/files#diff-9223cd6a0f3d50c7e3bd990b85d1c09fR41